### ISSUE TITLE
[BUG] Dataset statistics fail to load for larger datasets

### ISSUE THREAD

**Issue by brimoor** [voxel51/fiftyone#936]
There appears to be a wide variation in the time required for dataset statistics to load in the fields sidebar for larger datasets.

When working with the validation split of [bdd100k](https://voxel51.com/docs/fiftyone/user_guide/dataset_zoo/datasets.html#bdd100k), for example, I find that the loading icons in the screenshot below will persist for as little as 1-2 seconds to as many as 10-15 seconds or more. In the latter case, I typically click the cube to refresh the page, which will typically cause the stats to load faster if I try it a couple times.

This problem is not observed on datasets that are an order of magnitude smaller, such as [quickstart](https://voxel51.com/docs/fiftyone/user_guide/dataset_zoo/datasets.html#quickstart).

<img width="1108" alt="Screen Shot 2021-03-29 at 9 11 26 AM" src="https://user-images.githubusercontent.com/25985824/112842121-39c65180-906f-11eb-8a27-643d206512d7.png">

**Comment by benjaminpkane** [comment]
These statistics are taking between 4 and 5 seconds for me. Never 10-15 seconds.

You can get a sense of speeds with the below, although it seems like `asyncio` adds some overhead. A regular `pymongo` cursor yields ~3 seconds for me.

```py
import eta.core.utils as etau

import fiftyone as fo
import fiftyone.core.state as fos
import fiftyone.types as fot

bdd = fo.Dataset.from("/path/to/bdd/validation", dataset_type=fot.BDDDataset)

stats = fos.DatasetStatistics(bdd)
aggs = stats.aggregations + stats.exists_aggregations

with etau.Timer():
    bdd.aggregate(aggs)
```

I can't see things taking longer than ~5 seconds. At that point the response will likely never resolve, which I am also looking into. It might be best to take this offline tomorrow morning.

**Comment by benjaminpkane** [comment]
Resolved by #926
